### ISSUE TITLE
Android 6.0 IllegalArgumentException use HeldCertificate

### ISSUE THREAD

**Issue by crossle** [square/okhttp#6210]
I use the  HeldCertificate parse the key only on Android 6.x
```
    val heldCertificate = HeldCertificate.Builder().keyPair(this).build()
    return heldCertificate.privateKeyPkcs1Pem()
```

Fatal Exception: java.lang.IllegalArgumentException: Unknown pattern character 'X'
       at java.text.SimpleDateFormat.validatePatternCharacter(SimpleDateFormat.java:323)
       at java.text.SimpleDateFormat.validatePattern(SimpleDateFormat.java:312)
       at java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:365)
       at java.text.SimpleDateFormat.<init>(SimpleDateFormat.java:258)
       at okhttp3.tls.internal.der.Adapters.formatUtcTime$okhttp_tls(Adapters.java:177)
       at okhttp3.tls.internal.der.Adapters$UTC_TIME$1.encode(Adapters.java:158)
       at okhttp3.tls.internal.der.Adapters$UTC_TIME$1.encode(Adapters.java:152)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:84)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:26)
       at okhttp3.tls.internal.der.DerWriter.write(DerWriter.java:57)
       at okhttp3.tls.internal.der.BasicDerAdapter.toDer(BasicDerAdapter.java:83)
       at okhttp3.tls.internal.der.CertificateAdapters$time$1.toDer(CertificateAdapters.java:68)
       at okhttp3.tls.internal.der.CertificateAdapters$time$1.toDer(CertificateAdapters.java:43)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1$encode$1.invoke(Adapters.java:288)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1$encode$1.invoke(Adapters.java:265)
       at okhttp3.tls.internal.der.DerWriter.withTypeHint(DerWriter.java:102)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1.encode(Adapters.java:285)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:84)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:26)
       at okhttp3.tls.internal.der.DerWriter.write(DerWriter.java:57)
       at okhttp3.tls.internal.der.BasicDerAdapter.toDer(BasicDerAdapter.java:83)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1$encode$1.invoke(Adapters.java:288)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1$encode$1.invoke(Adapters.java:265)
       at okhttp3.tls.internal.der.DerWriter.withTypeHint(DerWriter.java:102)
       at okhttp3.tls.internal.der.Adapters$sequence$codec$1.encode(Adapters.java:285)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:84)
       at okhttp3.tls.internal.der.BasicDerAdapter$toDer$1.invoke(BasicDerAdapter.java:26)
       at okhttp3.tls.internal.der.DerWriter.write(DerWriter.java:57)
       at okhttp3.tls.internal.der.BasicDerAdapter.toDer(BasicDerAdapter.java:83)
       at okhttp3.tls.internal.der.DerAdapter$DefaultImpls.toDer(DerAdapter.java:58)
       at okhttp3.tls.internal.der.BasicDerAdapter.toDer(BasicDerAdapter.java:26)
       at okhttp3.tls.HeldCertificate$Builder.build(HeldCertificate.java:366)

**Comment by swankjesse** [comment]
Yikes. Can fix!

**Comment by swankjesse** [comment]
Fix released as OkHttp 4.8.1.
